tests/TestCIT_KCI.py

```diff
@@ -0,0 +1,290 @@
+                           0.578, 0.818, 0.589, 0.816, 0.578, 0.811, 0.596, 0.816, 0.578, 0.818, 0.601, 0.816, 0.578,
+                           0.822, 0.578, 0.816, 0.578, 0.822, 0.583, 0.816, 0.578, 0.812, 0.592, 0.816, 0.578, 0.812,
+                           0.578, 0.816, 0.578, 0.806, 0.578, 0.816, 0.578, 0.821, 0.591, 0.816, 0.578, 0.824, 0.584,
+                           0.816, 0.816, 0.813, 0.818, 0.816, 0.816, 0.803, 0.804, 0.816, 0.816, 0.824, 0.808, 0.816,
+                           0.816, 0.819, 0.816, 0.816, 0.816, 0.813, 0.81, 0.816, 0.816, 0.823, 0.801, 0.816, 0.816,
+                           0.804, 0.818, 0.816, 0.816, 0.825, 0.811, 0.816, 0.816, 0.817, 0.81, 0.816, 0.816, 0.812,
+                           0.808, 0.816, 0.816, 0.815, 0.816, 0.816, 0.816, 0.819, 0.817, 0.816, 0.816, 0.821, 0.824,
+                           0.816, 0.816, 0.811, 0.819, 0.816, 0.816, 0.815, 0.808, 0.816, 0.816, 0.812, 0.809, 0.816,
+                           0.816, 0.812, 0.812, 0.816, 0.816, 0.81, 0.807]
+
+        self.assertEqual(pvalue01, pvalue01_truth)
+        self.assertEqual(pvalue03, pvalue03_truth)
+        self.assertEqual(pvalue032, pvalue032_truth)
+
+    def test_Exponential_dist(self):
+        np.random.seed(50)
+        X = np.random.exponential(size=(300, 1))
+        X_prime = np.random.exponential(size=(300, 1))
+        Y = X + 0.5 * np.random.exponential(size=(300, 1))
+        Z = Y + 0.5 * np.random.exponential(size=(300, 1))
+        data = np.hstack((X, X_prime, Y, Z))
+
+        pvalue01 = []
+        pvalue03 = []
+        pvalue032 = []
+        for kernelname in ['Gaussian', 'Polynomial', 'Linear']:
+            for est_width in ['empirical', 'median', 'manual']:
+                for kwidth in [0.05, 0.1, 0.2]:
+                    for use_gp in [True, False]:
+                        for approx in [True, False]:
+                            for polyd in [1, 2]:
+                                cit_CIT = cit.CIT(data, 'kci', kernelX=kernelname, kernelY=kernelname,
+                                                  kernelZ=kernelname, est_width=est_width, use_gp=use_gp, approx=approx,
+                                                  polyd=polyd, kwidthx=kwidth, kwidthy=kwidth, kwidthz=kwidth)
+
+                                pvalue01.append(round(cit_CIT(0, 1), 3))
+                                # X and X_prime are independent, pvalue01 should be expected larger than 0.01
+                                pvalue03.append(round(cit_CIT(0, 3), 3))
+                                # X and Z are dependent, pvalue03 should be expected smaller than 0.01
+                                pvalue032.append(round(cit_CIT(0, 3, {2}), 3))
+                                # X and Z are independent conditional on Y, pvalue032 should be expected larger than 0.01
+
+        pvalue01_truth = [0.001, 0.001, 0.005, 0.007, 0.001, 0.001, 0.004, 0.001, 0.001, 0.001, 0.003, 0.006, 0.001,
```

Comment:
this sounds weird. How did you get the pvalue01_truth here?

We should expect > 0.01, but here lots of them are smaller than 0.01?

Did you check the result after computation?